src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -1445,49 +1445,33 @@ public override void Flush()
         internal sealed class ControlCHandlerRegistrar
         {
             private bool _handlerRegistered;
+            private IDisposable? _sigIntRegistration;
```

Comment:
Not that it makes a massive difference or anything, but you could avoid boxing and interface calls by just making these `PosixSignalRegistration`.